packages/kbn-plugin-helpers/tasks/build/create_build.js

```diff
@@ -83,6 +99,48 @@ module.exports = function createBuild(
         options
       );
     })
+    .then(function() {
+      const buildConfigPath = join(buildRoot, 'tsconfig.json');
+
+      if (!existsSync(buildConfigPath)) {
+        return;
+      }
+
+      if (
+        !plugin.pkg.devDependencies.typescript &&
+        !plugin.pkg.dependencies.typescript
```

Comment:
I regularly don't want to separate my dependencies between dev and prod because I'm not writing a library. If I was writing a Kibana plugin myself and I didn't have to worry about NOTICE.txt or anything I would put everything in production deps and never worry about specifying `--dev` at install or figuring out which deps to include when.